src/Rules/Functions/InvalidLexicalVariablesInClosureUseRule.php

```diff
@@ -0,0 +1,87 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Functions;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use function in_array;
+use function is_string;
+use function sprintf;
+
+/**
+ * @implements Rule<Node\Expr\Closure>
+ */
+class InvalidLexicalVariablesInClosureUseRule implements Rule
+{
+
+	private const SUPERGLOBAL_NAMES = [
+		'_COOKIE',
+		'_ENV',
+		'_FILES',
+		'_GET',
+		'_POST',
+		'_REQUEST',
+		'_SERVER',
+		'_SESSION',
+		'GLOBALS',
+	];
+
+	public function getNodeType(): string
+	{
+		return Node\Expr\Closure::class;
+	}
+
+	/**
+	 * @param Node\Expr\Closure $node
+	 */
+	public function processNode(Node $node, Scope $scope): array
+	{
+		$errors = [];
+		$params = [];
+
+		foreach ($node->getParams() as $param) {
+			if (!$param->var instanceof Node\Expr\Variable) {
+				continue;
+			}
+
+			if (!is_string($param->var->name)) {
+				continue;
+			}
+
+			$params[] = $param->var->name;
+		}
+
+		foreach ($node->uses as $use) {
+			if (!is_string($use->var->name)) {
+				continue;
+			}
+
+			$var = $use->var->name;
+
+			if ($var === 'this') {
+				$errors[] = RuleErrorBuilder::message('Cannot use $this as lexical variable.')->nonIgnorable()->build();
```

Comment:
To mark the exact line in multiline closure declarations, please use `->line` on the RuleErrorBuilder. In all reported errors here. Thanks.